src/networking.c

```diff
@@ -3559,564 +3620,681 @@ void quitCommand(client *c) {
-            "PAUSE <timeout> [WRITE|ALL]",
-            "    Suspend all, or just write, clients for <timeout> milliseconds.",
-            "REPLY (ON|OFF|SKIP)",
-            "    Control the replies sent to the current connection.",
-            "SETNAME <name>",
-            "    Assign the name <name> to the current connection.",
-            "SETINFO <option> <value>",
-            "    Set client meta attr. Options are:",
-            "    * LIB-NAME: the client lib name.",
-            "    * LIB-VER: the client lib version.",
-            "UNBLOCK <clientid> [TIMEOUT|ERROR]",
-            "    Unblock the specified blocked client.",
-            "TRACKING (ON|OFF) [REDIRECT <id>] [BCAST] [PREFIX <prefix> [...]]",
-            "         [OPTIN] [OPTOUT] [NOLOOP]",
-            "    Control server assisted client side caching.",
-            "TRACKINGINFO",
-            "    Report tracking status for the current connection.",
-            "NO-EVICT (ON|OFF)",
-            "    Protect current client connection from eviction.",
-            "NO-TOUCH (ON|OFF)",
-            "    Will not touch LRU/LFU stats when this mode is on.",
-            "IMPORT-SOURCE (ON|OFF)",
-            "    Mark this connection as an import source if import-mode is enabled.",
-            "    Sync tools can set their connections into 'import-source' state to visit",
-            "    expired keys.",
-            NULL};
-        addReplyHelp(c, help);
-    } else if (!strcasecmp(c->argv[1]->ptr, "id") && c->argc == 2) {
-        /* CLIENT ID */
-        addReplyLongLong(c, c->id);
-    } else if (!strcasecmp(c->argv[1]->ptr, "info") && c->argc == 2) {
-        /* CLIENT INFO */
-        sds o = catClientInfoString(sdsempty(), c, 0);
-        o = sdscatlen(o, "\n", 1);
-        addReplyVerbatim(c, o, sdslen(o), "txt");
-        sdsfree(o);
-    } else if (!strcasecmp(c->argv[1]->ptr, "list")) {
-        /* CLIENT LIST */
-        int type = -1;
-        sds o = NULL;
-        if (c->argc == 4 && !strcasecmp(c->argv[2]->ptr, "type")) {
-            type = getClientTypeByName(c->argv[3]->ptr);
-            if (type == -1) {
-                addReplyErrorFormat(c, "Unknown client type '%s'", (char *)c->argv[3]->ptr);
-                return;
+        if (!strcasecmp(c->argv[i]->ptr, "id")) {
+            if (filter->ids == NULL) {
+                filter->ids = intsetNew(); // Initialize the intset for IDs
             }
-        } else if (c->argc > 3 && !strcasecmp(c->argv[2]->ptr, "id")) {
-            int j;
-            o = sdsempty();
-            for (j = 3; j < c->argc; j++) {
-                long long cid;
-                if (getLongLongFromObjectOrReply(c, c->argv[j], &cid, "Invalid client ID")) {
-                    sdsfree(o);
-                    return;
+            i++; // Move to the first ID after "ID"
+
+            // Process all IDs until a non-numeric argument or end of args
```

Comment:
There used to be a preference for multi line comments. But I've not seen clang format check complaining about it.

Reply:
@zuiderkwast Do you know why it was enforced earlier though?

Reply:
Have changed few comments!